techniques/fileDistribution/checkGenericFileContent/8.0/checkGenericFileContent.st

```diff
@@ -0,0 +1,568 @@
+      "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_na", "${generic_file_content_uuid[${index}]}", "Post-modification hook", "${generic_file_content_path[${index}]}", "The file ${generic_file_content_path[${index}]} does not exists and is not configured to be created", "${index}"),
+        ifvarclass => "file_nothing_to_do_${index}";
+
+      "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_na", "${generic_file_content_uuid[${index}]}", "Post-modification hook", "${generic_file_content_path[${index}]}", "No command for ${generic_file_content_path[${index}]} was to be executed", "${index}"),
+        ifvarclass => "!execute_command_${index}.!file_nothing_to_do_${index}";
+
+      "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_success", "${generic_file_content_uuid[${index}]}", "Post-modification hook", "${generic_file_content_path[${index}]}", "${generic_file_content_path[${index}]} was in the proper state, so no command to execute", "${index}"),
+        ifvarclass => "execute_command_${index}.file_edition_global_files_status_${index}_RudderUniqueID_kept.!file_edition_global_files_status_${index}_RudderUniqueID_repaired";
+
+      "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_success", "${generic_file_content_uuid[${index}]}", "Post-modification hook", "${generic_file_content_path[${index}]}", "The command for ${generic_file_content_path[${index}]} was correctly executed", "${index}"),
+        ifvarclass => "generic_file_content_posthook_${index}_RudderUniqueID_command_run_ok";
+
+      "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_error", "${generic_file_content_uuid[${index}]}", "Post-modification hook", "${generic_file_content_path[${index}]}", "The command for ${generic_file_content_path[${index}]} could not to be executed", "${index}"),
+        ifvarclass => "generic_file_content_posthook_${index}_RudderUniqueID_command_run_error";
+
+      "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_success", "${generic_file_content_uuid[${index}]}", "Post-modification hook", "${generic_file_content_path[${index}]}", "${generic_file_content_path[${index}]} already exists, skipping command execution", "${index}"),
+        ifvarclass => "execute_command_${index}.creation_only_but_already_exists_${index}";
+
+  commands:
+    pass2::
+      "${generic_file_content_posthook[${index}]}"
+        classes    => if_else("generic_file_content_posthook_${index}_RudderUniqueID_command_run_ok", "generic_file_content_posthook_${index}_RudderUniqueID_command_run_error"),
+        contain    => in_shell,
+        ifvarclass => "execute_command_${index}.file_edition_global_files_status_${index}_RudderUniqueID_repaired";
+
+}
+
+
+# The file edition should be unitary made in this bundle
+bundle edit_line check_generic_file_content_edition_RudderUniqueID(
```

Comment:
i'd really like to put this in the rudder librairy, but it would be a fairly big change for only one change request